### PR TITLE
Feat fetch enemies

### PR DESCRIPTION
**Pull request by titi0267**
Fetch enemies, spawn enemies & fix clock for update

### CODE REVIEW

File: common/ressources/Map.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+#include <fstream>
+#include <iostream>
+
+namespace RType::Map {
+    class Map {
```
Comment (AlexandreFRANTZ) [inline common/ressources/Map.hpp:8]: Add doxygen

File: common/ressources/MapManager.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+#include <memory>
+#include <unordered_map>
+#include "Map.hpp"
+#include "../UDPHandler.hpp"
+#include "RuntimeException.hpp"
+#include "../../ecs/Coordinator.hpp"
+
+namespace RType::Map {
+    class MapManager {
```
Comment (AlexandreFRANTZ) [inline common/ressources/MapManager.hpp:11]: Add doxygen

File: common/ressources/RuntimeException.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <exception>
+#include <string>
+
+namespace RType::Map {
+    class RuntimeException : public std::exception {
```
Comment (AlexandreFRANTZ) [inline common/ressources/RuntimeException.hpp:7]: Add doxygen

File: server/MapManager.hpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+#include <memory>
+#include <unordered_map>
+#include <iostream>
+
+#include "../common/packet/SpawnEntity.hpp"
+#include "./Types.hpp"
+
+using namespace rapidjson;
+namespace RType {
+    class MapManager {
```
Comment (AlexandreFRANTZ) [inline server/MapManager.hpp:17]: Add doxygen

**Review by AlexandreFRANTZ — CHANGES_REQUESTED** [review]
Add all doxygen documentation

**Review by ludovic-str — CHANGES_REQUESTED** [review]
Clang-format

**Review by AlexandreFRANTZ — APPROVED** [review]

**Review by ludovic-str — APPROVED** [review]

**Pull request merged**